src/main/java/pl/sda/climbing_shop/controller/ClimbingGearController.java

```diff
@@ -0,0 +1,134 @@
+package pl.sda.climbing_shop.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import pl.sda.climbing_shop.category.Category;
+import pl.sda.climbing_shop.category.CategoryRepository;
+import pl.sda.climbing_shop.customer.Customer;
+import pl.sda.climbing_shop.customer.CustomerRepository;
+import pl.sda.climbing_shop.product.Product;
+import pl.sda.climbing_shop.product.ProductRepository;
+import pl.sda.climbing_shop.review.Review;
+import pl.sda.climbing_shop.review.ReviewRepository;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/climbingGear")
+public class ClimbingGearController {
+
+    private final ProductRepository productRepository;
+    private final CategoryRepository categoryRepository;
+    private final ReviewRepository reviewRepository;
+    private final CustomerRepository customerRepository;
+
+    @GetMapping
+    public String viewClimbingGear() {
+        return "climbingGear";
+    }
+
+    @GetMapping("/{categoryName}")
+    public String viewClimbingGearCategory(@PathVariable("categoryName") String categoryName, Model model,
+                                           @RequestParam(required = false) String type,
+                                           @RequestParam(required = false) String size,
+                                           @RequestParam(required = false) String color,
+                                           @RequestParam(required = false) String brand) {
+
+        return viewProducts(categoryName, model, type, size, color, brand, this.categoryRepository, this.productRepository);
```

Comment:
moze warto porworzyc jakies Viewmodele  na takie metody?
Mam na mysli jakas klase ktora to wszystko bedzie trzymac